### PR TITLE
AC_WPNav: protect from divide by zero especially when using CH6 tuning of WP speed

### PR DESCRIPTION
**Pull request by rmackay9**
This PR includes a few cosmetic fixes to AC_WPNav:

- correct spelling of "terrain"
- move definitions from .h to .cpp

.. and one real fix to avoid a divide-by-zero if ch6 tuning is being used to change the WP speed inflight.  The reason the divide-by-zero could happen was that channel 6 tuning would call AC_WPNav::set_speed_xy() before AC_WPNav::wp_and_spline_init() was called.  This means that _wp_desired_speed_xy_cms was still zero and led to the divide-by-zero.

This has been tested in SITL both before and after this change.  Below is a screenshot of the ch6 tuning knob being used to set the WPNAV_SPEED value.  Before this fix the simulator would crash and report the divide-by-zero.
image

This issue was discovered as part of the Copter-4.1.0-beta report, "Ch6 tuning of WP speed broken" on the [4.1 issues list](https://github.com/ArduPilot/ardupilot/issues/16478) and was reported [here on the forums](https://discuss.ardupilot.org/t/wp-speed-on-in-flight-tuning-not-functioning/75226).

### CODE REVIEW

File: libraries/AC_WPNav/AC_WPNav.h [hunk 1]
```diff
@@ -262,8 +250,8 @@ class AC_WPNav
     Vector3f    _origin;                // starting point of trip to next waypoint in cm from ekf origin
     Vector3f    _destination;           // target destination in cm from ekf origin
     float       _track_scalar_dt;       // time compression multiplier to slow the progress along the track
-    float       _terain_vel;            // maximum horizontal velocity used to ensure the aircraft can maintain height above terain
-    float       _terain_accel;          // acceleration value used to change _terain_vel
+    float       _terrain_vel;            // maximum horizontal velocity used to ensure the aircraft can maintain height above terrain
+    float       _terrain_accel;          // acceleration value used to change _terain_vel
```
Comment (peterbarker) [inline libraries/AC_WPNav/AC_WPNav.h:254]:
```suggestion
    float       _terrain_accel;          // acceleration value used to change _terrain_vel
```

Missed one :-)

**Review by peterbarker — APPROVED** [review]
LGTM

File: libraries/AC_WPNav/AC_WPNav.h (hunk 1 above)
Comment (rmackay9) [reply libraries/AC_WPNav/AC_WPNav.h]: cool, txs fixed.

**Comment by khancyr** [comment]
We should take this opportunity to change the defines for stztic const. That is far better on debugging

**Comment by peterbarker** [comment]
On Fri, 3 Sep 2021, Pierre Kancir wrote:

> We should take this opportunity to change the defines for stztic const. That is far better on debugging

I was going to mention that.  Then I thought, "if they're used in one 
place, just use them in one place".  Then I thought, "this is better" and 
approved it.

**Comment by rmackay9** [comment]
Thanks for the review and comments.  Let's merge this after Leonard's PR https://github.com/ArduPilot/ardupilot/pull/18531 goes in.

**Comment by rmackay9** [comment]
Leonard's PR is in so merging this one too.

**Pull request merged**